x-pack/test/api_integration/apis/ml/jobs/reset.ts

```diff
@@ -0,0 +1,148 @@
+import { ANNOTATION_TYPE } from '@kbn/ml-plugin/common/constants/annotations';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { MULTI_METRIC_JOB_CONFIG, SINGLE_METRIC_JOB_CONFIG, DATAFEED_CONFIG } from './common_jobs';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const testSetupJobConfigs = [SINGLE_METRIC_JOB_CONFIG, MULTI_METRIC_JOB_CONFIG];
+
+  async function runResetJobsRequest(
+    user: USER,
+    requestBody: object,
+    expectedResponsecode: number
+  ): Promise<any> {
+    const { body, status } = await supertest
+      .post('/api/ml/jobs/reset_jobs')
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send(requestBody);
+    ml.api.assertResponseStatusCode(expectedResponsecode, status, body);
+
+    return body;
+  }
+
+  async function createAnnotation(jobId: string, annotation: string) {
+    await ml.api.indexAnnotation({
+      timestamp: 1549756524346,
+      end_timestamp: 1549766472273,
+      annotation,
+      job_id: jobId,
+      type: ANNOTATION_TYPE.ANNOTATION,
+      detector_index: 0,
+      event: 'user',
+    });
+  }
+
+  const expectedResetResponseBody = {
+    [SINGLE_METRIC_JOB_CONFIG.job_id]: { reset: true, task: 'cannot be predicted' },
+    [MULTI_METRIC_JOB_CONFIG.job_id]: { reset: true, task: 'cannot be predicted' },
+  };
+
+  describe('reset_jobs', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.createIndexPatternIfNeeded('ft_farequote', '@timestamp');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+
+      for (const job of testSetupJobConfigs) {
+        await ml.api.deleteAnomalyDetectionJobES(job.job_id);
+      }
+      await ml.api.cleanMlIndices();
+    });
+
+    after(async () => {
+      await ml.testResources.deleteIndexPatternByTitle('ft_farequote');
+    });
+
+    beforeEach(async () => {
+      for (const job of testSetupJobConfigs) {
+        const jobId = job.job_id;
+        const datafeedId = `datafeed-${jobId}`;
+        await ml.api.createAnomalyDetectionJob(job);
+        await ml.api.openAnomalyDetectionJob(jobId);
+        await ml.api.createDatafeed({
+          ...DATAFEED_CONFIG,
+          datafeed_id: datafeedId,
+          job_id: jobId,
+        });
+
+        await ml.api.startDatafeed(datafeedId, { start: '0', end: `${Date.now()}` });
+        await ml.api.waitForDatafeedState(datafeedId, DATAFEED_STATE.STOPPED);
+        await ml.api.waitForJobState(jobId, JOB_STATE.CLOSED);
```

Comment:
I think this block could be replaced by a `ml.api.createAndRunAnomalyDetectionLookbackJob` call.

Reply:
updated in https://github.com/elastic/kibana/pull/150880/commits/1ad4e0c6922c24360cdcd3349e2151ac566c743d